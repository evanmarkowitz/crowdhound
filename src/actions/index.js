export const toggleFilterModal = boolean => ({
  type: 'TOGGLE_FILTER_MODAL',
  boolean
});

// export const setFilterCriteria = criteria => ({
//   type: 'SET_FILTER_CRITERIA',
//   criteria
// });

export const setDistanceValue = distance => ({
  type: 'SET_DISTANCE_VALUE',
  distance
});

export const setActivityLevel = activeLevel => ({
  type: 'SET_ACTIVE_LEVEL',
  activeLevel
});

export const setDogSize = size => ({
  type: 'SET_DOG_SIZE',
  size
});