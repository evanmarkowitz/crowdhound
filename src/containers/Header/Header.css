.header {
  display: flex;
  height: 60px;
  align-items: center;
  width: 355px;
  justify-content: space-around;
  background: rgb(243, 243, 243);
  position: fixed;
  top: 0;
  left: 0;
  margin-top: 20px;
}

.logo {
  height: 50%;
  margin-right: 15px;
  margin-bottom: 6px;
}

.burger-container {
  width: 32px;
  height: 25px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  position: relative;
}

.burger-btn {
  position: absolute;
  top: 0;
  bottom: 0;
  height: 100%;
  width: 100%;
  background: none;
  cursor: pointer;
  outline: none;
  border: none;
}

.burger-line {
  width: 100%;
  height: 5px;
  background: gray;
  border-radius: 3px;
}



.burger-container:hover .burger-line {
  background: #1dbbdf;
}