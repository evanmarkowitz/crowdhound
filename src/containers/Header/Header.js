import React, { Component } from 'react';
import './Header.css';
import logo from '../../images/crowd-hound-logo-2.png';
import { connect } from 'react-redux';
import { toggleFilterModal } from '../../actions';
import { Link } from 'react-router-dom';

export class Header extends Component {

  toggleBurger = () => {
    const {toggleFilterModal,  toggleFilterValue} = this.props
    toggleFilterModal(!toggleFilterValue)
  }

  activeBurgerStyle = () => {
    const {toggleFilterValue} = this.props;
    return toggleFilterValue ? {background: '#1dbbdf'} : {background: 'gray'}
  }

  render() {
    return (
      <header className="header">
        <section className="burger-container" >
            <button className="burger-btn" onClick={this.toggleBurger}></button>
            <div style={this.activeBurgerStyle()} className="burger-line burger-line-1"></div>
            <div style={this.activeBurgerStyle()} className="burger-line burger-line-2"></div>
            <div style={this.activeBurgerStyle()} className="burger-line burger-line-3"></div>
        </section>
        <Link to="/" className="logo"><img src={logo} alt="logo"  className="logo-inside"/></Link>
      </header>
    )
  }
}

export const mapStateToProps = state => ({
  toggleFilterValue:  state.toggleFilterValue,
})

export const mapDispatchToProps = dispatch => ({
  toggleFilterModal: boolean => dispatch(toggleFilterModal(boolean)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Header)