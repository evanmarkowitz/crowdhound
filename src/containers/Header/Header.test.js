import React from 'react';
import { Header, mapDispatchToProps} from './Header';
import { shallow } from 'enzyme';
import { toggleFilterModal } from '../../actions';




describe('Header', () => {
  let wrapper;
  let props;
  beforeEach(() => {
    props = {
      toggleFilterModal: jest.fn(),
      toggleFilterValue: true
    }
     wrapper = shallow(<Header {...props}/>);
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('toggleBurger should run toggleFilterModal function', () => {
    wrapper.instance().toggleBurger();
    expect(props.toggleFilterModal).toHaveBeenCalled()
  })

  it('should dispatch toggleFilterModal action when toggleFilterModal is called', () => {
    const mockDispatch = jest.fn();
    const mockSelected = true;
    const mockAction = toggleFilterModal(mockSelected);
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.toggleFilterModal(mockSelected);
    expect(mockDispatch).toHaveBeenCalledWith(mockAction);
  })

  it.skip('activeBurgerStyle should return the correct style', () => {
    console.log('xxxx',props.toggleFilterValue)
    const spy = jest.spyOn(wrapper.instance(), "activeBurgerStyle");
    wrapper.instance().forceUpdate();
    expect(spy).toHaveBeenCalledWith(1)

  })

})
