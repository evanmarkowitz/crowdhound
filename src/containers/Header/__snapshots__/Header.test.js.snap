// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot 1`] = `ShallowWrapper {}`;

exports[`Header should match the snapshot 2`] = `ShallowWrapper {}`;
